vault/core.go

```diff
@@ -4280,3 +4280,46 @@ func (c *Core) GetRaftAutopilotState(ctx context.Context) (*raft.AutopilotState,
 func (c *Core) Events() *eventbus.EventBus {
 	return c.events
 }
+
+func (c *Core) SetSeals(barrierSeal Seal, secureRandomReader io.Reader) error {
```

Comment:
Method `SetSeals` looks like a method that should be called every time the barrier seal is set on the core object. However, the only caller in this PR is the new `ServerCommand.reloadSeals` method. This asymmetry makes me think that either the code that sets the seal normally should be calling this method, or if that is not appropriate, then perhaps this method should really just be part of the reloading code?

Reply:
It looks like the seal is normally set when creating the core, so I don't think it could use this function. I think this is only relevant in the reload case, but it does need to be on the `Core`, because it needs to edit private fields